src/SDKs/Azure.Base/data-plane/Azure.Base/FailedResponseException.cs

```diff
@@ -7,15 +7,28 @@ namespace Azure
 {
     public class RequestFailedException : Exception
     {
-        Response _response;
-        string _reason;
+        int _status;
 
-        public RequestFailedException(Response response, string reason = null)
+        public RequestFailedException(Response response, string summary = null)
+            : this(response, summary, null)
+        {}
+
+        public RequestFailedException(Response response, string summary, Exception innerException)
+            : base(FormatMessage(response, summary), innerException)
+        {
+            _status = response.Status;
+        }
+
+        private static string FormatMessage(Response response, string summary)
         {
-            _response = response;
-            _reason = reason;
+            if (summary != null) {
+                return $"{summary}\n{response}";
```

Comment:
Should we consider `Environment.NewLine` over `\n` here for a better cross-platform format?

Reply:
Good point. Will change.